core/state/journal.go

```diff
@@ -166,7 +166,12 @@ func (ch resetObjectChange) dirtied() *common.Address {
 }
 
 func (ch suicideChange) revert(s *StateDB) {
-	obj := s.getStateObject(*ch.account)
+	var obj *StateObject
+	if s.parallel.isSlotDB {
```

Comment:
we should abstract this into one function

Reply:
Yes, I will try to remove this code.

Reply:
Yes, I will try to remove this code.